plugins/polio/api/campaigns/campaigns.py

```diff
@@ -258,10 +258,19 @@ def update(self, instance: Campaign, validated_data):
             # we pop the campaign since we use the set afterward which will also remove the deleted one
             round_data.pop("campaign", None)
             scopes = round_data.pop("scopes", [])
+
+            # # Replace ReasonForDelay instance with key_name to avoid type error when calling is_valid
```

Comment:
```suggestion
            # Replace ReasonForDelay instance with key_name to avoid type error when calling is_valid
```

BTW, not sure to understand this comment. Let's talk about it.

Reply:
Because the serializer is nested, the data is transformed several times, and the `key_name` is replaced with `name_en` I think. In an earlier version I had the same problem with a PrimaryKeyRelatedField (I think), where the pk was replaced by the instance at the validation step. When I passed the data to the serializer, I got type error because it expected a pk an received the instance instead. Same general idea here.